benches/pool.rs

```diff
@@ -26,14 +26,58 @@ mod remem {
     }
 
     fn run(thread: usize, iter: usize) {
-        let p = Pool::new(|| vec![0usize; CAPACITY]);
+        let p = Pool::new(|| vec![0u8; CAPACITY]);
         let mut threads = Vec::new();
 
         for _ in 0..thread {
             let p = p.clone();
             threads.push(thread::spawn(move || {
                 for _ in 0..iter {
-                    let mut v = p.get();
+                    let mut v = black_box(p.get());
+                    v[0] = 1;
+                    v[CAPACITY / 4] = 1;
+                    v[CAPACITY / 2] = 1;
+                }
+            }));
+        }
+
+        for t in threads {
+            t.join().unwrap();
+        }
+    }
+}
+
+mod byte_pool {
+    use super::CAPACITY;
+    use byte_pool::BytePool;
+    use std::sync::Arc;
+    use std::thread;
+    use test::{black_box, Bencher};
+
+    #[bench]
+    fn create(b: &mut Bencher) {
+        b.iter(|| Arc::new(BytePool::new()));
```

Comment:
this is measuring pool creation, not actual object allocations

Reply:
oh yeah, that's what it's intended to measure haha.

Reply:
Following async-std's lead here: https://github.com/async-rs/async-std/blob/master/benches/mutex.rs#L9-L12

Reply:
In that case I would argue this should be set to `b.iter(|| {})` for the `Vec` case, as it has no overhead, as there is no pool being created

Reply:
Yea I'm also not sure that the create benches all measure the same thing. Some are fully initialized and some are not